perun-core/src/main/java/cz/metacentrum/perun/core/blImpl/AuthzResolverBlImpl.java

```diff
@@ -170,13 +170,11 @@ private static Boolean doBeforeAttributeRightsCheck(PerunSession sess, ActionTyp
 			return true;
 		}
 
-		// Engine and Service can read attributes
-		if (sess.getPerunPrincipal().getRoles().hasRole(Role.ENGINE) && actionType.equals(ActionType.READ)) {
-			return true;
-		}
-
-		// RPC can read attributes
-		if (sess.getPerunPrincipal().getRoles().hasRole(Role.RPC) && actionType.equals(ActionType.READ)) {
+		// Engine, Service, RPC and Perunobserver can read attributes
+		if (actionType.equals(ActionType.READ) &&
```

Comment:
Please allow also other types of READ (VO/PUBLIC) for these three roles. Thank you.